---
title: "Violin plots, KM curves and UMAP coloring"
output: html_notebook
author: Nayanga Thirimanne
date: "November 2025"
---

```{r}
library(tidyverse)
library(reshape)
library(ggplot2)
library(rstatix)
library(RColorBrewer)
library(dplyr)
install.packages("ggsurvfit")
library(ggsurvfit)
install.packages("survminer")
library(survminer)
library(ggpubr)
library(patchwork)

```

Plot parameters
```{r}
plot_title_size = 20
legend_pt_size = 4
axis_text_size = 25
axis_title_size = 25
legend_text_size = 15
spacing = 1
chosen_margin = c(0.5,1,0.5,1) #top,right,bottom,left
theme_grid_nolegend <- theme_bw()+
  theme(plot.title = element_text(hjust=0, vjust=0, lineheight=.8, face="bold", size=plot_title_size),
        plot.margin = unit(chosen_margin,"cm"),
        legend.text = element_text(size=legend_text_size),
        legend.key.height = unit(spacing,"cm"),
        legend.position = "none",
        legend.justification = "left",
        legend.title = element_blank())
font_size = 18
theme_grid_legend <- theme_bw()+
  theme(plot.title = element_text(hjust=0, vjust=0, lineheight=.8, face="bold", size=plot_title_size),
        legend.text = element_text(size=legend_text_size, face = "bold"),
        legend.key.height = unit(spacing,"cm"),
        legend.position = "top",
        legend.justification = "left",
        legend.title = element_blank()
        )

theme_legend <- theme_void()+
  theme(plot.title = element_text(hjust=0, vjust=0, lineheight=.8, face="bold", size=plot_title_size),
        plot.margin = unit(chosen_margin,"cm"),
        legend.text = element_text(size=legend_text_size, face = "bold"),
        legend.key.height = unit(spacing,"cm"),
        legend.position = "right",
        legend.justification = "left",
        legend.title = element_blank() )

theme_nolegend <- theme_void()+
  theme(plot.title = element_text(hjust=0, vjust=0, lineheight=.8, face="bold", size=plot_title_size),
        plot.margin = unit(chosen_margin,"cm"),
        legend.text = element_text(size=legend_text_size),
        legend.key.height = unit(spacing,"cm"),
        legend.position = "none",
        legend.justification = "left",
        legend.title = element_blank() )


theme_KM <- theme_bw()+
  theme(plot.title = element_text(hjust=0, vjust=0, lineheight=.8, face="bold", size=plot_title_size),
        plot.margin = unit(chosen_margin,"cm"),
        legend.text = element_text(size=legend_text_size, face = "bold"),
        legend.key.height = unit(spacing,"cm"),
        legend.position = "top",
        legend.justification = "left",
        legend.title = element_blank() ,
        axis.text = element_text(size = axis_text_size_km, face = "bold", color = "black"),
        axis.title = element_text(size = axis_title_size_km, face = "bold", color = "black"))
     
```

Load data
```{r}
#meningioma umap coordinates
mening_umap2d <- read.csv("~/HollandLabShared/Meningioma_bulkrnaseq_datasets/Meningioma_CellGenomics_finalfiles/1298_vst_umap2dcoord.csv", check.names = F)

#meningioma clinical data
sample_info <-read.csv("~/HollandLabShared/Meningioma_bulkrnaseq_datasets/Meningioma_CellGenomics_finalfiles/Meningioma_clinical_patient_v61_20240430.csv", check.names = F)
finaldf <- left_join(mening_umap2d, sample_info, by="coordinate_ID")

#meningioma normalized count
mening_vst <- readRDS("~/HollandLabShared/Nayanga/Meningioma_Eric/Output_VSTcounts/batchcor_vstcounts_1298.rds")
mening_tpm <- readRDS("~/HollandLabShared/Nayanga/Meningioma_Eric/1298samples_combatseq_log2tpm.rds")

#copy number calls (from casper analysis in Thirimanne et al.)
menin_CNV <- read.csv("~/HollandLabShared/Nayanga/Meningioma_Eric/Analysis17_CNV/1298_CNVcalls.csv", check.names = F) %>% .[,-c(1, 47,48)]
finaldf_CNV <- left_join(mening_umap2d, menin_CNV, by="coordinate_ID")
```

```{r}
setwd("~/HollandLabShared/Nayanga/Abby/")
```

```{r}
#cluster A = Aggressive NF2 mutant region
cluster_A <- subset(finaldf, finaldf$DBSCAN_CLUSTER == "A")
FF3 <- subset(finaldf_CNV, finaldf_CNV$`10q` == -1) 
A10qloss <- subset(cluster_A, cluster_A$coordinate_ID %in% FF3$coordinate_ID)
A10qintact <- subset(cluster_A, !(cluster_A$coordinate_ID %in% A10qloss$coordinate_ID))

#cluster B = Benign NF2 mutant region
cluster_B <- subset(finaldf, finaldf$DBSCAN_CLUSTER == "B")

clusterAB <- subset(finaldf, (finaldf$DBSCAN_CLUSTER == "A" | finaldf$DBSCAN_CLUSTER == "B"))
AB10qloss <- subset(clusterAB, clusterAB$coordinate_ID %in% FF3$coordinate_ID)
AB10qintact <- subset(clusterAB, !(clusterAB$coordinate_ID %in% AB10qloss$coordinate_ID))
```

PTEN levels in Chr10q loss/intact in NF2 mutant region (Agg. and Benign)
```{r}
mencl <- as.data.frame(finaldf[, c("coordinate_ID")]) 
colnames(mencl)[1] <- "coordinate_ID"
mencl$AB10qloss[mencl$coordinate_ID %in% AB10qloss$coordinate_ID] <- "AB10qloss"
mencl$AB10qintact[mencl$coordinate_ID %in% AB10qintact$coordinate_ID] <- "AB10qintact"

clusterIDs <- mencl[, c("coordinate_ID", "AB10qloss", "AB10qintact")]

gene = "PTEN"
CC <- pivot_longer(data = clusterIDs, cols =  c("AB10qloss", "AB10qintact")) %>% .[,-2] %>% na.omit(.)
colnames(CC)[2] <- "cluster"

VV <- mening_tpm %>%
as.data.frame(.) %>%
  dplyr::filter(., rownames(.) %in% gene) %>%
  t(.)  %>%
  as.data.frame(.) %>%
  rownames_to_column(., var = "coordinate_ID") %>%
  left_join(., CC, by = "coordinate_ID") %>% 
  na.omit(.)

colnames(VV)[2] <- "value"

VV %>% 
  ggplot(aes(x = cluster, y = value, fill = cluster)) +
  geom_violin( alpha = 0.5, draw_quantiles = c(0.25, 0.5, 0.75)) +
  ylab("Counts") +
  xlab("Cluster") +
  scale_fill_manual(values = c("grey55", "blue")) +
  stat_compare_means(method = "wilcox.test", label = "p.format", 
                     comparisons = list(c("AB10qintact", "AB10qloss")),
                     label.y = max(VV$value) * 1.05) +
  theme_grid_nolegend +
  theme(axis.title.x = element_text(size = 15, face = "bold")) +
  theme(axis.title.y = element_text(size = 15, face = "bold")) +
  theme(axis.text.x = element_text(size = 15, face = "bold", color = "black")) +
  theme(axis.text.y = element_text(size = 15, face = "bold", color = "black")) +
  theme(strip.text = element_text(size = 15, face = "bold")) 

ggsave("violin_clusterAB_chr10q_PTEN.pdf", height = 6, width=6)

```

t.test
```{r}
cl_AB10qloss <- subset(VV, cluster == "AB10qloss")
cl_AB10qintact <- subset(VV, cluster == "AB10qintact")

t_test_A10q <- t.test(cl_AB10qintact$value, cl_AB10qloss$value)
t_test_A10q
```

PTEN levels in Chr10q loss/intact in Aggressive NF2 mutant region
```{r}
mencl <- as.data.frame(finaldf[, c("coordinate_ID")]) 
colnames(mencl)[1] <- "coordinate_ID"
mencl$A10qloss[mencl$coordinate_ID %in% A10qloss$coordinate_ID] <- "A10qloss"
mencl$A10qintact[mencl$coordinate_ID %in% A10qintact$coordinate_ID] <- "A10qintact"

clusterIDs <- mencl[, c("coordinate_ID", "A10qloss", "A10qintact")]

gene = "PTEN"
CC <- pivot_longer(data = clusterIDs, cols =  c("A10qloss", "A10qintact")) %>% .[,-2] %>% na.omit(.)
colnames(CC)[2] <- "cluster"

VV <- mening_tpm %>%
as.data.frame(.) %>%
  filter(., rownames(.) %in% gene) %>%
  t(.)  %>%
  as.data.frame(.) %>%
  rownames_to_column(., var = "coordinate_ID") %>%
  left_join(., CC, by = "coordinate_ID") %>% 
  na.omit(.)

colnames(VV)[2] <- "value"

VV %>% 
  ggplot(aes(x = cluster, y = value, fill = cluster)) +
  geom_violin( alpha = 0.5, draw_quantiles = c(0.25, 0.5, 0.75)) +
  ylab("Counts") +
  xlab("Cluster") +
  scale_x_discrete(limits=c( "A10qintact", "A10qloss")) +
  scale_fill_manual(values = c("grey55", "blue")) +
  stat_compare_means(method = "wilcox.test", label = "p.format", 
                     comparisons = list(c("A10qloss", "A10qintact")),
                     label.y = max(VV$value) * 1.05) +
  theme_grid_nolegend +
  theme(axis.title.x = element_text(size = 15, face = "bold")) +
  theme(axis.title.y = element_text(size = 15, face = "bold")) +
  theme(axis.text.x = element_text(size = 15, face = "bold", color = "black")) +
  theme(axis.text.y = element_text(size = 15, face = "bold", color = "black")) +
  theme(strip.text = element_text(size = 15, face = "bold")) 

setwd("~/HollandLabShared/Nayanga/Abby/")
ggsave("violin_clusterA_chr10q_PTEN.pdf", height = 6, width=6)
```


PTEN levels in Chr10q loss/intact in Aggressive NF2 mutant, Benign NF2 mutant and NF2-Wild Type regions
```{r}

mencl <- as.data.frame(finaldf[, c("coordinate_ID")]) 
colnames(mencl)[1] <- "coordinate_ID"
mencl$cluster[mencl$coordinate_ID %in% finaldf$coordinate_ID[finaldf$DBSCAN_CLUSTER == "A"]] <- "Agg.NF2mut."
mencl$cluster[mencl$coordinate_ID %in% finaldf$coordinate_ID[finaldf$DBSCAN_CLUSTER == "B"]] <- "Ben.NF2mut."
mencl$cluster[mencl$coordinate_ID %in% finaldf$coordinate_ID[finaldf$DBSCAN_CLUSTER == "C"]] <- "NF2WT"
mencl$cluster[mencl$coordinate_ID %in% finaldf$coordinate_ID[finaldf$DBSCAN_CLUSTER == "D"]] <- "NF2WT"


clusterIDs <- mencl[, c("coordinate_ID", "cluster")]

gene = "PTEN"
CC <- pivot_longer(data = clusterIDs, cols =  c("cluster")) %>% .[,-2] %>% na.omit(.)
colnames(CC)[2] <- "cluster"


VV <- mening_tpm %>%
as.data.frame(.) %>%
  filter(., rownames(.) %in% gene) %>%
  t(.)  %>%
  as.data.frame(.) %>%
  rownames_to_column(., var = "coordinate_ID") %>%
  left_join(., CC, by = "coordinate_ID") %>% 
  na.omit(.)

colnames(VV)[2] <- "value"

VV %>% 
  ggplot(aes(x = cluster, y = value, fill = cluster)) +
  geom_violin( alpha = 0.5, draw_quantiles = c(0.25, 0.5, 0.75)) +
  ylab("PTEN expression") +
 scale_x_discrete(limits=c("NF2WT", "Ben.NF2mut.",  "Agg.NF2mut.")) +
  scale_fill_manual(values = c("red", "forestgreen","blue")) +
  stat_compare_means(method = "wilcox.test", label = "p.format", 
                     comparisons = list(c("NF2WT", "Ben.NF2mut."), 
                                        c("Ben.NF2mut.", "Agg.NF2mut."), 
                                        c("NF2WT", "Agg.NF2mut.")
                                        ),
    label.y = c(
    max(VV$value) * 1.05,
    max(VV$value) * 1.10,
    max(VV$value) * 1.15,
    max(VV$value) * 1.20)) +
  theme_grid_nolegend +
  theme(axis.title.x = element_text(size = 15, face = "bold")) +
  theme(axis.title.y = element_text(size = 15, face = "bold")) +
  theme(axis.text.x = element_text(size = 15, face = "bold", color = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y = element_text(size = 15, face = "bold", color = "black")) +
  theme(strip.text = element_text(size = 15, face = "bold")) 


setwd("~/HollandLabShared/Nayanga/Abby/")
ggsave("violin_PTEN_meningioma_3clusters.pdf", height = 6, width=6)


```

```{r}
mencl <- as.data.frame(finaldf[, c("coordinate_ID")]) 
colnames(mencl)[1] <- "coordinate_ID"
mencl$cluster[mencl$coordinate_ID %in% A10qloss$coordinate_ID] <- "A10qloss"
mencl$cluster[mencl$coordinate_ID %in% A10qintact$coordinate_ID] <- "A10qintact"
mencl$cluster[mencl$coordinate_ID %in% finaldf$coordinate_ID[finaldf$DBSCAN_CLUSTER == "B"]] <- "Ben.NF2mut."
mencl$cluster[mencl$coordinate_ID %in% finaldf$coordinate_ID[finaldf$DBSCAN_CLUSTER == "C"]] <- "NF2WT"
mencl$cluster[mencl$coordinate_ID %in% finaldf$coordinate_ID[finaldf$DBSCAN_CLUSTER == "D"]] <- "NF2WT"


clusterIDs <- mencl[, c("coordinate_ID", "cluster")]

gene = "PTEN"
CC <- pivot_longer(data = clusterIDs, cols =  c("cluster")) %>% .[,-2] %>% na.omit(.)
colnames(CC)[2] <- "cluster"


VV <- mening_tpm %>%
as.data.frame(.) %>%
  filter(., rownames(.) %in% gene) %>%
  t(.)  %>%
  as.data.frame(.) %>%
  rownames_to_column(., var = "coordinate_ID") %>%
  left_join(., CC, by = "coordinate_ID") %>% 
  na.omit(.)

colnames(VV)[2] <- "value"

VV %>% 
  ggplot(aes(x = cluster, y = value, fill = cluster)) +
  geom_violin( alpha = 0.5, draw_quantiles = c(0.25, 0.5, 0.75)) +
  ylab("PTEN expression") +
 scale_x_discrete(limits=c("NF2WT", "Ben.NF2mut.",  "A10qintact", "A10qloss")) +
  scale_fill_manual(values = c("grey", "blue","grey",  "grey")) +
  stat_compare_means(method = "wilcox.test", label = "p.format", 
                     comparisons = list(c("NF2WT", "Ben.NF2mut."), 
                                        c("Ben.NF2mut.", "A10qintact"), 
                                        c("NF2WT", "A10qintact"),
                                        c("NF2WT", "A10qloss"),
                                        c("Ben.NF2mut.", "A10qloss"),
                                        c("A10qintact", "A10qloss")
                                        ),
    label.y = c(
    max(VV$value) * 1.05,
    max(VV$value) * 1.10,
    max(VV$value) * 1.15,
    max(VV$value) * 1.20)) +
  theme_grid_nolegend +
  theme(axis.title.x = element_text(size = 15, face = "bold")) +
  theme(axis.title.y = element_text(size = 15, face = "bold")) +
  theme(axis.text.x = element_text(size = 15, face = "bold", color = "black", angle = 45, hjust = 1)) +
  theme(axis.text.y = element_text(size = 15, face = "bold", color = "black")) +
  theme(strip.text = element_text(size = 15, face = "bold")) 


setwd("~/HollandLabShared/Nayanga/Abby/")
ggsave("violin_PTEN_suppfigure2A.pdf", height = 6, width=6)

```

